lib/utils/aws/aws.go

```diff
@@ -142,7 +149,7 @@ func GetAndReplaceReqBody(req *http.Request) ([]byte, error) {
 
 // drainBody drains the body, close the reader and returns the read bytes.
 func drainBody(b io.ReadCloser) ([]byte, error) {
-	payload, err := io.ReadAll(b)
+	payload, err := utils.ReadAtMost(b, teleport.MaxHTTPRequestSize)
```

Comment:
We should read request bodies using the methods provided to prevent resource exhaustion attacks.

Reply:
You've changed the semantics of the method so that it no longer drains the body. This may or may not be correct (I don't have the context). But if this is a change we want to keep, I would rename the func to something more accurate.

I'm not sure what the purpose of this is. In general, I typically see _response_ bodies drained, which ensures that the HTTP server is able to reuse existing connections rather than create new ones. I'm not sure that's necessary on request bodies though.

Reply:
I can change it to `tryDrainBody`. We should probably also change it to defer the body close so that it always closes the body at least.

drainBody is called here:
```
// GetAndReplaceReqBody returns the request and replace the drained body reader with io.NopCloser
// allowing for further body processing by http transport.
func GetAndReplaceReqBody(req *http.Request) ([]byte, error) {
	if req.Body == nil || req.Body == http.NoBody {
		return []byte{}, nil
	}
	// req.Body is closed during drainBody call.
	payload, err := drainBody(req.Body)
	if err != nil {
		return nil, trace.Wrap(err)
	}

	// Replace the drained body with io.NopCloser reader allowing for further request processing by HTTP transport.
	req.Body = io.NopCloser(bytes.NewReader(payload))
	return payload, nil
}
```

We need to read the request body in order to sign the request in `lib/srv/app/aws/handler.SigningService`. But after reading the request body we need to replace it in order to send the request.

Reply:
Updated